web-common/src/features/canvas-components/render/VegaRenderer.svelte

```diff
@@ -0,0 +1,153 @@
+<script lang="ts">
+  import CancelCircle from "@rilldata/web-common/components/icons/CancelCircle.svelte";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { get } from "svelte/store";
+  import {
+    SignalListeners,
+    Vega,
+    View,
+    VisualizationSpec,
+    type EmbedOptions,
+  } from "svelte-vega";
+  import { ExpressionFunction, VLTooltipFormatter } from "../types";
+  import { VegaLiteTooltipHandler } from "./vega-tooltip";
+  import { onDestroy } from "svelte";
+  import { getRillTheme } from "./vega-config";
+
+  export let data: Record<string, unknown> = {};
+  export let spec: VisualizationSpec;
+  export let signalListeners: SignalListeners = {};
+  export let expressionFunctions: ExpressionFunction = {};
+  export let error: string | null = null;
+  export let canvasDashboard = false;
+  export let chartView = false;
```

Comment:
It seems a little off for a "Vega[Lite]Renderer" to be aware of canvas dashboards. I suppose I'd expect that  a "Vega[Lite]Renderer" would solely render Vega visualizations, and any Rill application logic (e.g. references to canvases or the runtime) would be handled outside of it.

(Though I noticed that the `VegaLiteRenderer` also includes a `canvasDashboard` prop, so I do agree with keeping the two Vega renderers symmetrical.)

Reply:
https://rilldata.slack.com/archives/C04AGHL77PS/p1726514541389759?thread_ts=1726509359.103919&cid=C04AGHL77PS